File: 0122. Best Time to Buy and Sell Stock II.js
// INPUT: [ 7, 1, 5, 3, 6, 4 ]
// OUTPUT: 7

const maxProfit = (prices) => {
  let i = (profit = 0);
  while (i < prices.length - 1) {
    while (i < prices.length - 1 && prices[i] >= prices[i + 1]) i++;
    profit -= prices[i];
    while (i < prices.length - 1 && prices[i] <= prices[i + 1]) i++;
    profit += prices[i];
  }

  return profit;
};

console.log(maxProfit([7, 1, 5, 3, 6, 4]));

File: python/0122. Best Time to Buy and Sell Stock II.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0

        for idx in range(1, len(prices)):
            if prices[idx - 1] < prices[idx]:
                profit += prices[idx] - prices[idx - 1]

        return profit
